### ISSUE TITLE
NoPlayerFoundException on Android 10

### ISSUE THREAD

**Issue by Faiyyaz** [florent37/Flutter-AssetsAudioPlayer#284]
**Flutter Version**

My version : ">=2.7.0 <3.0.0"

**Lib Version**

My version : ^2.0.7+9

**Platform (Android / iOS / web) + version**

Platform : Android 10

**Describe the bug**

When i play the sound it gives me below error on android

```
W/System.err( 8438): com.github.florent37.assets_audio_player.playerimplem.PlayerFinder$NoPlayerFoundException
W/System.err( 8438): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder._findWorkingPlayer(PlayerFinder.kt:68)
W/System.err( 8438): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder._findWorkingPlayer(PlayerFinder.kt:86)
W/System.err( 8438): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder$_findWorkingPlayer$1.invokeSuspend(Unknown Source:12)
W/System.err( 8438): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/System.err( 8438): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:55)
W/System.err( 8438): 	at android.os.Handler.handleCallback(Handler.java:883)
W/System.err( 8438): 	at android.os.Handler.dispatchMessage(Handler.java:100)
W/System.err( 8438): 	at android.os.Looper.loop(Looper.java:238)
W/System.err( 8438): 	at android.app.ActivityThread.main(ActivityThread.java:7817)
W/System.err( 8438): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 8438): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
W/System.err( 8438): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:981)
```

**Small code to reproduce**

```dart
import 'package:assets_audio_player/assets_audio_player.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State {
  
  final Audio audio = Audio('assets/audios/refresh.mp3');
  final AssetsAudioPlayer _assetsAudioPlayer = AssetsAudioPlayer();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: RaisedButton(
            child: Text("open"),
            onPressed: () {
              //open code here
              _playAudio();
            }
          ),
        ),
      ),
    );
  }

 _playAudio() {
    _assetsAudioPlayer.open(audio);
    _assetsAudioPlayer.play();
  }

@override
  void dispose() {
    super.dispose();
    _assetsAudioPlayer.stop();
    _assetsAudioPlayer.dispose();
  }
}
```

**Comment by florent37** [comment]
NoPlayerFoundException happen when the audio is unplayable,
can you send me the mp3 ? maybe it has a specific encoding ?

**Comment by Faiyyaz** [comment]
PFA mp3 file in ZIP format


[refresh.zip](https://github.com/florent37/Flutter-AssetsAudioPlayer/files/4911880/refresh.zip)

**Comment by daniellampl** [comment]
I encountered the exact same problem when trying to play with audios from network.

**Comment by florent37** [comment]
@daniellampl on android10 ?

**Comment by daniellampl** [comment]
Yes, on an Oneplus 5. When I open the player with assets or local files everything works fine.

**Comment by AlexV525** [comment]
Encounting the same problem.

Media served with content Uri: `content::/media/external/audio/media/1578145` .

I debugged a few methods, turns out it throws an error in `_findWorkingPlayer` .

https://github.com/florent37/Flutter-AssetsAudioPlayer/blob/14229fc8f21ed58cf663ae045ef07c2888f87c54/android/src/main/kotlin/com/github/florent37/assets_audio_player/playerimplem/PlayerFinder.kt#L73-L75

When the code running into this section, it throws an `NetworkError` which is confusing: Why would I need networking for a content Uri? (See the screenshot)

![image](https://user-images.githubusercontent.com/15884415/87397238-9d464700-c5e6-11ea-8fdf-2c6c8868f37b.png)

After that it ran into the `_findWorkingPlayer` recursively and throws the `NoPlayerFoundException` .

**Comment by florent37** [comment]
oh because it's an uri like : content::/media/external/audio/media/1578145

I think it needs a content provider for this, don't remember the implemntation

for now files are like /sdcard/directoory1/file.mp3

**Comment by florent37** [comment]
I have to see if exoplayer can read this directly or if it needs more setup

**Comment by AlexV525** [comment]
> oh because it's an uri like : content::/media/external/audio/media/1578145

Sorry about the typo: `content://media` .

Before it works on `2.0.6+x` .

**Comment by leonicolas** [comment]
Same problem here...
This is the stacktrace:

`W/System.err( 4494): com.github.florent37.assets_audio_player.playerimplem.PlayerFinder$NoPlayerFoundException
W/System.err( 4494): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder._findWorkingPlayer(PlayerFinder.kt:68)
W/System.err( 4494): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder._findWorkingPlayer(PlayerFinder.kt:86)
W/System.err( 4494): 	at com.github.florent37.assets_audio_player.playerimplem.PlayerFinder$_findWorkingPlayer$1.invokeSuspend(Unknown Source:12)
W/System.err( 4494): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/System.err( 4494): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:55)
W/System.err( 4494): 	at android.os.Handler.handleCallback(Handler.java:873)
W/System.err( 4494): 	at android.os.Handler.dispatchMessage(Handler.java:99)
W/System.err( 4494): 	at android.os.Looper.loop(Looper.java:193)
W/System.err( 4494): 	at android.app.ActivityThread.main(ActivityThread.java:6669)
W/System.err( 4494): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 4494): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
W/System.err( 4494): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)`

**Comment by florent37** [comment]
I added some code to fix #283 maybe it has fixed it, can you try 2.0.8+3 and reopen if it does not work ?

**Comment by jk7d1e** [comment]
For me it worked to change the audio to ``` final Audio audio = Audio.file('assets/audios/refresh.mp3');```. I think its a background processing thing from @florent37 's side :)
